### DISCUSSION TITLE
VS Code extended keywords

### DISCUSSION THREAD

**Discussion by michaeltlombardi** [hyperjump-io/json-schema-language-tools#113] [General]
👋🏻 I took a quick peek at building and trying out the prototype extension locally - I've been working extensively on a couple projects that requires authoring JSON Schemas in 2020-12 and found the limitations in VS Code pretty painful for developing schemas newer than draft 7.

One thing I've found rather _nice_ in VS Code is the experience you get authoring/editing JSON that _has_ a defined schema using the (largely undocumented [annotation keywords VS Code recognizes][01], like `markdownDescription` and `patternErrorMessage`):

I'm wondering whether support for that functionality (or an equivalent, using a defined/published dialect that the extension can recognize) is in _eventual_ scope.

[01]: https://github.com/microsoft/vscode-json-languageservice/blob/1b453f9ab5e3f6bba7c5f16b9821e871b7bc8298/src/jsonSchema.ts#L77-L87

**Comment by jdesrosiers** [comment]
Yes, _eventually_ that kind of thing will be in scope, but there's a lot more to get through before then. For now, the focus is on features for authoring JSON Schemas as opposed to authoring JSON that's described by a JSON Schema which is what VSCode supports. So, at least for now, this isn't a replacement for VSCode's JSON features. Ideally, VSCode steps up and updates to support 2020-12 and this language server will be purely complementary, but if they don't, I'll make a replacement. Eventually. I'm work on this in my spare time now, so progress is slow.

**Reply by michaeltlombardi** [reply]
Thank you for clarifying, and for starting on this work- I'm already finding this _really_ useful for authoring and validating my schemas themselves. I also have some projects that involve defining vocabulary, and I'm finding this useful in that process. I agree about the ideal outcome, especially as the extant support from the built-in language service is halfway to 2020-12 support already.

**Reply by jdesrosiers** [reply]
Thanks for being an early adopter. I'm glad you're finding it useful even at this early stage.
